File: go/leap/leap.go
// Package leap enables the identification of leap years on Gregorian calendar dates.
package leap

// IsLeapYear returns true to a Gregoria calendar leap year, false otherwise.
func IsLeapYear(year int) bool {
	leapYear := year%4 == 0
	roundCenturyYear := year%100 == 0
	leapCentury := year%400 == 0

	return leapYear && !roundCenturyYear || leapCentury
}

File: fortran/leap/leap.f90
module leap
  implicit none

contains

  logical function is_leap_year(year)
    integer :: year

    if (mod(year, 100) == 0) then
      is_leap_year = mod(year, 400) == 0
    else
      is_leap_year = mod(year, 4) == 0 
    endif
  end function

end module

